src/app/qgisapp.cpp

```diff
@@ -14428,15 +14637,22 @@ void QgisApp::layersWereAdded( const QList<QgsMapLayer *> &layers )
       provider = vProvider;
     }
 
-    QgsRasterLayer *rlayer = qobject_cast<QgsRasterLayer *>( layer );
-    if ( rlayer )
+    if ( QgsRasterLayer *rlayer = qobject_cast<QgsRasterLayer *>( layer ) )
     {
       // connect up any request the raster may make to update the statusbar message
       connect( rlayer, &QgsRasterLayer::statusChanged, this, &QgisApp::showStatusMessage );
 
       provider = rlayer->dataProvider();
     }
 
+    if ( QgsMeshLayer *mlayer = qobject_cast<QgsMeshLayer *>( layer ) )
+    {
+      connect( mlayer, &QgsMeshLayer::layerModified, this, &QgisApp::updateLayerModifiedActions );
```

Comment:
This could be simplified now -- instead of setting up the connections in layer type specific blocks, you could instead move these connections out of the blocks and create them regardless of the layer type.

Reply:
damn, of course ...